datageneration/retrieve_combinations.py

```diff
@@ -368,13 +371,14 @@ def generate_tag_list_with_properties(self) -> List[TagCombination]:
     args = parser.parse_args()
 
     source = args.source
-    prop_limit = args.prop_limit
+    prop_limit = int(args.prop_limit)
```

Comment:
You don't have to convert int, you can add type field to parser.add_argument and define type=int for the int properties.